components/script/dom/webglrenderingcontext.rs

```diff
@@ -167,31 +167,16 @@ impl WebGLRenderingContext {
     }
 
     #[allow(unsafe_code)]
-    fn float32_array_to_slice(&self, data: *mut JSObject) -> Option<Vec<f32>> {
+    fn array_buffer_view_to_vec<T: Clone>(&self, data: *mut JSObject) -> Option<Vec<T>> {
```

Comment:
The `to_vec()` is necessary in the context of WebGL since it's going to be sent over the WebGL task, but not needed otherwise.

I updated the PR, and used it in the other sites where `JS_GetObjectAsArrayBufferView` was being called.